vaapi/vaapiimagepool.cpp

```diff
@@ -69,6 +69,7 @@ struct VaapiImagePool::ImageRecycler
         if (!image)
             return;
         m_pool->recycleID(image->getID());
+        m_pool.reset();
```

Comment:
Are you sure this will help you ? Vfter void operator()(VaapiImage* image) called, the ImageRecycler will out of scope.It will reset m_pool automatically. It should have same result no matter you reset here or not.